From 3b6e61af3454e79c712b94c9fbd623b22b32e44e Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Sat, 3 Apr 2021 14:01:42 +0000
Subject: [PATCH] KDLConfig.cmake.in: don't include absolute patch to
 Eigen_INCLUDE_DIR

* otherwise absolute path ends in exported orocos_kdl-config.cmake like this:
  $ grep eigen3 image/usr/share/orocos_kdl/cmake/orocos_kdl-config.cmake
  set(orocos_kdl_INCLUDE_DIRS "${CMAKE_CURRENT_LIST_DIR}/../../../include;;/jenkins/mjansa/build/ros/webos-noetic-hardknott/tmp-glibc/work/qemux86-webos-linux/orocos-kdl/1.4.0-r0/recipe-sysroot/usr/include/eigen3")

  and when e.g. mavros tries to use it, it fails, because orocos-kdl RSS
  might already be removed by rm_work

| CMake Error at /jenkins/mjansa/build/ros/webos-noetic-dunfell/tmp-glibc/work/qemux86-webos-linux/mavros/1.6.0-1-r0/recipe-sysroot/usr/opt/ros/noetic/share/eigen_conversions/cmake/eigen_conversionsConfig.cmake:123 (message):
|   Project 'eigen_conversions' specifies
|   '/jenkins/mjansa/build/ros/webos-noetic-dunfell/tmp-glibc/work/qemux86-webos-linux/orocos-kdl/1.4.0-r0/recipe-sysroot/usr/include/eigen3'
|   as an include dir, which is not found.  It does not exist in any of the
|   root filesystems.  Ask the maintainer 'Tully Foote
|   <tfoote@osrfoundation.org>' to fix it.
| Call Stack (most recent call first):
|   /jenkins/mjansa/build/ros/webos-noetic-dunfell/tmp-glibc/work/qemux86-webos-linux/mavros/1.6.0-1-r0/recipe-sysroot/usr/opt/ros/noetic/share/catkin/cmake/catkinConfig.cmake:76 (find_package)
|   CMakeLists.txt:7 (find_package)

* the @Boost_INCLUDE_DIRS@ is already empty, drop that as well

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 orocos_kdl/KDLConfig.cmake.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/orocos_kdl/KDLConfig.cmake.in b/orocos_kdl/KDLConfig.cmake.in
index 3c3097c..1dc1cf0 100644
--- a/orocos_kdl/KDLConfig.cmake.in
+++ b/orocos_kdl/KDLConfig.cmake.in
@@ -4,8 +4,10 @@
 #  orocos_kdl_LIBRARIES    - libraries to link against for Orocos KDL
 #  orocos_kdl_PKGCONFIG_DIR - directory containing the .pc pkgconfig files
 
+find_package (Eigen3 3.3 REQUIRED)
+
 # Compute paths
-set(orocos_kdl_INCLUDE_DIRS "${CMAKE_CURRENT_LIST_DIR}/../../../include;@Boost_INCLUDE_DIRS@;@Eigen_INCLUDE_DIR@")
+set(orocos_kdl_INCLUDE_DIRS "${CMAKE_CURRENT_LIST_DIR}/../../../include;${EIGEN3_INCLUDE_DIR}")
 
 if(NOT TARGET orocos-kdl)
   include("${CMAKE_CURRENT_LIST_DIR}/OrocosKDLTargets.cmake")
